# **Descripción del proyecto**

En este proyecto, trabajarás con datos reales de transmisión de música online para explorar y procesar información sobre los hábitos de escucha de los usuarios y las usuarias en dos ciudades: Springfield y Shelbyville.

# **Instrucciones para completar el proyecto**

Se te proporciona una plantilla del cuaderno, donde puedes escribir tu código y proporcionar descripciones de tu trabajo. Para completar el proyecto, rellena cada celda de código en la plantilla y edita las celdas Markdown donde la plantilla te pide que expliques tus resultados.

Siempre es una buena práctica incluir una introducción que describa brevemente tus objetivos y una conclusión que resuma tus resultados en forma de celdas Markdown. 

El proyecto se divide en dos etapas principales, cada una con objetivos específicos:

- Etapa 1: Descripción de los datos.


El cuaderno tiene celdas preparadas con instrucciones sobre qué tipo de código escribir, así como bloques de texto donde puedes anotar tus observaciones.

- Etapa 2: Preprocesamiento de datos.


En esta etapa, limpiarás los datos examinando los nombres de las columnas y abordando los valores duplicados y ausentes.
Sigue la estructura proporcionada en el cuaderno y asegúrate de anotar tus observaciones al final de esta sección.

- Etapa 3: Análisis.


En esta etapa, analizarás si la actividad de los usuarios y las usuarias presenta variaciones según el día de la semana y la ciudad.
Es el núcleo de tu proyecto.
Sigue los pasos de programación necesarios para probar cada declaración y comenta tus resultados en los bloques apropiados.
Finalmente, resume los resultados de todo el proyecto en la sección "Conclusiones".

# **Diccionario de datos**

Los datos se almacenan en el archivo /datasets/music_project_en.csv.

## **Descripción de las columnas:**

'userID': identifica de forma exclusiva a cada usuario o usuaria;

'Track': título de la canción;

'artist': nombre del artista;

'genre': género musical;

'City': ciudad del usuario o la usuaria;

'time': hora del día en la que se reprodujo la pista (HH:MM:SS);

'Day': día de la semana.

## **Introducción** <a id='intro'></a>

Como analista de datos, tu trabajo consiste en analizar datos para extraer información valiosa y tomar decisiones basadas en ellos. Esto implica diferentes etapas, como la descripción general de los datos, el preprocesamiento y la prueba de hipótesis.

Siempre que investigamos, necesitamos formular hipótesis que después podamos probar. A veces aceptamos estas hipótesis; otras veces, las rechazamos. Para tomar las decisiones correctas, una empresa debe ser capaz de entender si está haciendo las suposiciones correctas.

En este proyecto, compararás las preferencias musicales de las ciudades de Springfield y Shelbyville. Estudiarás datos reales de transmisión de música online para probar la hipótesis a continuación y comparar el comportamiento de los usuarios y las usuarias de estas dos ciudades.




### **1.1. Objetivo:**
Prueba la hipótesis:

La actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad.


### **Etapas**

Los datos del comportamiento del usuario se almacenan en el archivo `/datasets/music_project_en.csv`. No hay ninguna información sobre la calidad de los datos, así que necesitarás examinarlos antes de probar la hipótesis.

Primero, evaluarás la calidad de los datos y verás si los problemas son significativos. Entonces, durante el preprocesamiento de datos, tomarás en cuenta los problemas más críticos.

Tu proyecto consistirá en tres etapas:
 1. Descripción de los datos.
 2. Preprocesamiento de datos.
 3. Prueba de hipótesis.

## **Etapa 1. Descripción de los datos** <a id='data_review'></a>

Abre los datos y examínalos.

Necesitarás pandas, así que impórtalo.

In [23]:
# Importar pandas
import pandas as pd




In [24]:
import os
os.listdir(r"C:\Users\pedro\Desktop\datasets")

['music_project_en.csv', 'Proyecto Sprint 4.1']

In [27]:
df = pd.read_csv(
    r"C:\Users\pedro\Desktop\datasets\music_project_en.csv",
    encoding="latin1"
)

Lee el archivo music_project_en.csv de la carpeta /datasets/ y guárdalo en la variable df:

In [28]:
df = pd.read_csv(r"C:\Users\pedro\Desktop\datasets\music_project_en.csv", encoding="latin1")
df.head()

userID                        Track             artist  genre  \
0  FFB692EC            Kamigata To Boots   The Mass Missile   rock   
1  55204538  Delayed Because of Accident  Andreas Rönnberg   rock   
2    20EC38          Funiculì funiculà        Mario Lanza    pop   
3  A3DD03C9        Dragons in the Sunset         Fire + Ice   folk   
4  E2DC1FAE                  Soul People         Space Echo  dance   

        City        time        Day  
0  Shelbyville  20:28:33  Wednesday  
1  Springfield  14:07:09     Friday  
2  Shelbyville  20:58:07  Wednesday  
3  Shelbyville  08:37:09     Monday  
4  Springfield  08:34:34     Monday

Obtén la información general sobre la tabla con un comando. Conoces el método que muestra la información general que necesitamos.

In [29]:
# Obtener la información general sobre nuestros datos
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65079 entries, 0 to 65078
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0     userID  65079 non-null  object
 1   Track     63736 non-null  object
 2   artist    57512 non-null  object
 3   genre     63881 non-null  object
 4     City    65079 non-null  object
 5   time      65079 non-null  object
 6   Day       65079 non-null  object
dtypes: object(7)
memory usage: 3.5+ MB


Estas son nuestras observaciones sobre la tabla. Contiene siete columnas. Almacenan los mismos tipos de datos: object.

Según la documentación:

- ' userID': identificador del usuario o la usuaria;

- 'Track': título de la canción;

- 'artist': nombre del artista;

- 'genre': género de la pista;

- 'City': ciudad del usuario o la usuaria;

- 'time': la hora exacta en la que se reprodujo la canción;

- 'Day': día de la semana.

Podemos ver tres problemas con el estilo en los encabezados de la tabla:

1. Algunos encabezados están en mayúsculas, otros en minúsculas.

2. Hay espacios en algunos encabezados.

3. Detecta el tercer problema por tu cuenta y descríbelo aquí.

Respuesta: Lo que detecto es que podemos renombrar los nombres de la columna "time" y "Day" con la idea de que los encabezados resulten más descriptivos (explícitos).

### **Escribe observaciones de tu parte. Estas son algunas de las preguntas que pueden ser útiles:** <a id='data_review_conclusions'></a>

`1.   ¿Qué tipo de datos tenemos a nuestra disposición en las filas? ¿Y cómo podemos entender lo que almacenan las columnas?`

Datos de tipo "object". Podemos entender lo que almacenan las columnas como información relacionada con tipo de música esuchada en dos determinados lugares, así como para analizar el tráfico de usuarios en un determinado espacio de tiempo.

`2.   ¿Hay suficientes datos para proporcionar respuestas a nuestra hipótesis o necesitamos más información?`
# **Hipótesis: La actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad.**
Considero que sí, ya que con el tipo y cantidad de datos puede ser suficiente para dar alguna respuesta a la hipótesis, sea aceptada total o parcialmente, negada total o parcialmente.


`3.   ¿Notaste algún problema en los datos, como valores ausentes, duplicados o tipos de datos incorrectos?`

Afirmativo, esto puede ser visualizado al momento de solicitar la información general de los datos.

## **Etapa 2. Preprocesamiento de datos** <a id='data_preprocessing'></a>

El objetivo aquí es preparar los datos para que sean analizados.
El primer paso es resolver cualquier problema con los encabezados. Luego podemos avanzar a los valores ausentes y duplicados. Empecemos.

Corrige el formato en los encabezados de la tabla.

### **2.1 Estilo del encabezado** <a id='header_style'></a>
Muestra los encabezados de la tabla (los nombres de las columnas):

In [30]:
# Muestra los nombres de las columnas

print(df.columns)

Index(['  userID', 'Track', 'artist', 'genre', '  City  ', 'time', 'Day'], dtype='object')


Cambia los encabezados de la tabla de acuerdo con las reglas del buen estilo:
* Todos los caracteres deben ser minúsculas.
* Elimina los espacios.
* Si el nombre tiene varias palabras, utiliza snake_case.

Anteriormente, aprendiste acerca de la forma automática de cambiar el nombre de las columnas. Vamos a aplicarla ahora. Utiliza el bucle for para iterar sobre los nombres de las columnas y poner todos los caracteres en minúsculas. Cuando hayas terminado, vuelve a mostrar los encabezados de la tabla:

In [31]:
# Bucle en los encabezados poniendo todo en minúsculas

new_col_names=[]
for col in df.columns:
    col = col.lower()
    new_col_names.append(col)
    
df.columns=new_col_names

print(df.columns)

Index(['  userid', 'track', 'artist', 'genre', '  city  ', 'time', 'day'], dtype='object')


Ahora, utilizando el mismo método, elimina los espacios al principio y al final de los nombres de las columnas e imprime los nombres de las columnas nuevamente:

In [32]:
# Bucle en los encabezados eliminando los espacios

new_col_names=[]
for col in df.columns:
    #col = col.lower()
    col = col.strip()
    new_col_names.append(col)
    
df.columns=new_col_names
print(df.columns)

Index(['userid', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')


Necesitamos aplicar la regla de snake_case a la columna `userid`. Debe ser `user_id`. Cambia el nombre de esta columna y muestra los nombres de todas las columnas cuando hayas terminado.

In [34]:
# Cambiar el nombre de la columna "userid"

new_col_names=[]
for col in df.columns:
    #col = col.lower()
    #col = col.strip()
    col = col.replace("userid","user_id")
    new_col_names.append(col)
    
df.columns=new_col_names
print(df.columns)


Index(['user_id', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')


Comprueba el resultado. Muestra los encabezados una vez más:

In [36]:
# Comprobar el resultado: la lista de encabezados

print(new_col_names)

['user_id', 'track', 'artist', 'genre', 'city', 'time', 'day']


### **2.2 Valores ausentes** <a id='missing_values'></a>

 Primero, encuentra el número de valores ausentes en la tabla. Debes utilizar dos métodos en una secuencia para obtener el número de valores ausentes.

In [37]:
# Calcular el número de valores ausentes

   
df.columns.isna().sum()


print(df.isna().sum())

user_id       0
track      1343
artist     7567
genre      1198
city          0
time          0
day           0
dtype: int64


No todos los valores ausentes afectan a la investigación. Por ejemplo, los valores ausentes en `track` y `artist` no son cruciales. Simplemente puedes reemplazarlos con valores predeterminados como el string `'unknown'` (desconocido).

Pero los valores ausentes en `'genre'` pueden afectar la comparación entre las preferencias musicales de Springfield y Shelbyville. En la vida real, sería útil saber las razones por las cuales hay datos ausentes e intentar recuperarlos. Pero no tenemos esa oportunidad en este proyecto. Así que tendrás que:
* rellenar estos valores ausentes con un valor predeterminado;
* evaluar cuánto podrían afectar los valores ausentes a tus cómputos;

Reemplazar los valores ausentes en las columnas `'track'`, `'artist'` y `'genre'` con el string `'unknown'`. Como mostramos anteriormente en las lecciones, la mejor forma de hacerlo es crear una lista que almacene los nombres de las columnas donde se necesita el reemplazo. Luego, utiliza esta lista e itera sobre las columnas donde se necesita el reemplazo haciendo el propio reemplazo.

In [38]:
# Bucle en los encabezados reemplazando los valores ausentes con 'unknown'


col_to_replace=('track','artist','genre')

for col in col_to_replace:
    df[col]=df[col].fillna('unknown')
    


print(df.head(20))

     user_id                                              track  \
0   FFB692EC                                  Kamigata To Boots   
1   55204538                        Delayed Because of Accident   
2     20EC38                                Funiculì funiculà   
3   A3DD03C9                              Dragons in the Sunset   
4   E2DC1FAE                                        Soul People   
5   842029A1                                             Chains   
6   4CB90AA5                                               True   
7   F03E1C1F                                   Feeling This Way   
8   8FA1D3BE                                         L’estate   
9   E772D5C0                                          Pessimist   
10  BC5A3A29                                       Gool la Mita   
11  8B5192C0  Is There Anybody Out There? (Panoramic Paralys...   
12  FF3FD2BD                                              Truth   
13  CC782B0F                               After School Specia

### **2.3 Valores duplicados**

Encuentra el número de duplicados explícitos en la tabla. Una vez más, debes aplicar dos métodos en una secuencia para obtener la cantidad de duplicados explícitos.

In [39]:
# Contar duplicados explícitos
print(df.duplicated().sum())

3826


Ahora, elimina todos los duplicados. Para ello, llama al método que hace exactamente esto.

In [40]:
# Eliminar duplicados explícitos
df.drop_duplicates(inplace=True)

Comprobemos ahora si eliminamos con éxito todos los duplicados. Cuenta los duplicados explícitos una vez más para asegurarte de haberlos eliminado todos:

In [41]:
# Comprobar de nuevo si hay duplicados
print(df.duplicated().sum())

0


Ahora queremos deshacernos de los duplicados implícitos en la columna genre. Por ejemplo, el nombre de un género se puede escribir de varias formas. Dichos errores también pueden afectar al resultado.

Para hacerlo, primero mostremos una lista de nombres de género únicos, ordenados en orden alfabético. Para ello:

- Extrae la columna genre del DataFrame.

- Llama al método que devolverá todos los valores únicos en la columna extraída.


In [42]:
# Inspeccionar los nombres de géneros únicos
sorted(df['genre'].unique())

['acid',
 'acoustic',
 'action',
 'adult',
 'africa',
 'afrikaans',
 'alternative',
 'ambient',
 'americana',
 'animated',
 'anime',
 'arabesk',
 'arabic',
 'arena',
 'argentinetango',
 'art',
 'audiobook',
 'avantgarde',
 'axé',
 'baile',
 'balkan',
 'beats',
 'bigroom',
 'black',
 'bluegrass',
 'blues',
 'bollywood',
 'bossa',
 'brazilian',
 'breakbeat',
 'breaks',
 'broadway',
 'cantautori',
 'cantopop',
 'canzone',
 'caribbean',
 'caucasian',
 'celtic',
 'chamber',
 'children',
 'chill',
 'chinese',
 'choral',
 'christian',
 'christmas',
 'classical',
 'classicmetal',
 'club',
 'colombian',
 'comedy',
 'conjazz',
 'contemporary',
 'country',
 'cuban',
 'dance',
 'dancehall',
 'dancepop',
 'dark',
 'death',
 'deep',
 'deutschrock',
 'deutschspr',
 'dirty',
 'disco',
 'dnb',
 'documentary',
 'downbeat',
 'downtempo',
 'drum',
 'dub',
 'dubstep',
 'eastern',
 'easy',
 'electronic',
 'electropop',
 'emo',
 'entehno',
 'epicmetal',
 'estrada',
 'ethnic',
 'eurofolk',
 'european',
 'exp

Busca en la lista para encontrar duplicados implícitos del género hiphop. Estos pueden ser nombres escritos incorrectamente o nombres alternativos para el mismo género.

Verás los siguientes duplicados implícitos:

- hip

- hop

- hip-hop

Para deshacerte de ellos, crea una función llamada replace_wrong_genres() con dos parámetros:

- wrong_genres=: esta es una lista que contiene todos los valores que necesitas reemplazar.

- correct_genre=: este es un string que vas a utilizar como reemplazo.

Como resultado, la función debería corregir los nombres en la columna 'genre' de la tabla df, es decir, remplazar cada valor de la lista wrong_genres por el valor en correct_genre.

Dentro del cuerpo de la función, utiliza un bucle 'for' para iterar sobre la lista de géneros incorrectos, extrae la columna 'genre' y aplica el método replace para hacer correcciones.

In [43]:
# Función para reemplazar duplicados implícitos
def replace_wrong_genres (wrong_genres,correct_genre):
    for w in wrong_genres:
        df['genre']=df['genre'].replace(w,correct_genre)
        
    return df

Ahora, llama a replace_wrong_genres() y pásale tales argumentos para que retire los duplicados implícitos (hip, hop y hip-hop) y los reemplace por hiphop:

In [44]:
# Eliminar duplicados implícitos
new_list=['hip','hop','hip-hop']
replace_wrong_genres(new_list,'hiphop')

user_id                              track             artist  \
0      FFB692EC                  Kamigata To Boots   The Mass Missile   
1      55204538        Delayed Because of Accident  Andreas Rönnberg   
2        20EC38                Funiculì funiculà        Mario Lanza   
3      A3DD03C9              Dragons in the Sunset         Fire + Ice   
4      E2DC1FAE                        Soul People         Space Echo   
...         ...                                ...                ...   
65074  729CBB09                            My Name             McLean   
65075  D08D4A55  Maybe One Day (feat. Black Spade)        Blu & Exile   
65076  C5E3A0D5                          Jalopiina            unknown   
65077  321D0506                      Freight Train      Chas McDevitt   
65078  3A64EF84          Tell Me Sweet Little Lies       Monica Lopez   

            genre         city      time        day  
0            rock  Shelbyville  20:28:33  Wednesday  
1            rock  Springfield  14:07:09     Friday  
2             pop  Shelbyville  20:58:07  Wednesday  
3            folk  Shelbyville  08:37:09     Monday  
4           dance  Springfield  08:34:34     Monday  
...           ...          ...       ...        ...  
65074         rnb  Springfield  13:32:28  Wednesday  
65075      hiphop  Shelbyville  10:00:00     Monday  
65076  industrial  Springfield  20:09:26     Friday  
65077        rock  Springfield  21:43:59     Friday  
65078     country  Springfield  21:59:46     Friday  

[61253 rows x 7 columns]

Asegúrate de que los nombres duplicados han sido eliminados. Muestra la lista de valores únicos de la columna 'genre' una vez más:

In [45]:
# Comprobación de duplicados implícitos
sorted(df['genre'].unique())

['acid',
 'acoustic',
 'action',
 'adult',
 'africa',
 'afrikaans',
 'alternative',
 'ambient',
 'americana',
 'animated',
 'anime',
 'arabesk',
 'arabic',
 'arena',
 'argentinetango',
 'art',
 'audiobook',
 'avantgarde',
 'axé',
 'baile',
 'balkan',
 'beats',
 'bigroom',
 'black',
 'bluegrass',
 'blues',
 'bollywood',
 'bossa',
 'brazilian',
 'breakbeat',
 'breaks',
 'broadway',
 'cantautori',
 'cantopop',
 'canzone',
 'caribbean',
 'caucasian',
 'celtic',
 'chamber',
 'children',
 'chill',
 'chinese',
 'choral',
 'christian',
 'christmas',
 'classical',
 'classicmetal',
 'club',
 'colombian',
 'comedy',
 'conjazz',
 'contemporary',
 'country',
 'cuban',
 'dance',
 'dancehall',
 'dancepop',
 'dark',
 'death',
 'deep',
 'deutschrock',
 'deutschspr',
 'dirty',
 'disco',
 'dnb',
 'documentary',
 'downbeat',
 'downtempo',
 'drum',
 'dub',
 'dubstep',
 'eastern',
 'easy',
 'electronic',
 'electropop',
 'emo',
 'entehno',
 'epicmetal',
 'estrada',
 'ethnic',
 'eurofolk',
 'european',
 'exp

### **2.4 Tus Observaciones**

Describe brevemente lo que has notado al analizar duplicados, cómo abordaste sus eliminaciones y qué resultados obtuviste.

Los datos duplicados representan un 17% del total de datos. Las eliminaciones se abordaron primero identificando por medio del método duplicated() el número de valores duplicados. Después por medio del método drop() se procedió a eliminar esos valores. Posteriormente eliminar datos duplicados implícitos. Al final se obtuvo una lista limpia sin géneros duplicados.

## **Etapa 3. Prueba de hipótesis.**

### **2.1. Hipótesis: comparar el comportamiento del usuario o la usuaria en las dos ciudades**

La hipótesis afirma que existen diferencias en la forma en que los usuarios y las usuarias de Springfield y Shelbyville consumen música. Para comprobar esto, usa los datos de tres días de la semana: lunes, miércoles y viernes.

- Agrupa a los usuarios y las usuarias por ciudad.

- Compara el número de canciones que cada grupo reprodujo el lunes, el miércoles y el viernes.


Realiza cada cálculo por separado.

El primer paso es evaluar la actividad del usuario en cada ciudad. Recuerda las etapas dividir-aplicar-combinar de las que hablamos anteriormente en la lección. Tu objetivo ahora es agrupar los datos por ciudad, aplicar el método apropiado para contar durante la etapa de aplicación y luego encontrar la cantidad de canciones reproducidas en cada grupo especificando la columna para obtener el recuento.

A continuación se muestra un ejemplo de cómo debería verse el resultado final: df.groupby(by='....')['column'].method()Realiza cada cálculo por separado.

Para evaluar la actividad de los usuarios y las usuarias en cada ciudad, agrupa los datos por ciudad y encuentra la cantidad de canciones reproducidas en cada grupo.

In [46]:
# Contar las canciones reproducidas en cada ciudad
#groupby()

musica_por_ciudad=df.groupby(by='city')['track'].count().reset_index(name='reproducciones')
musica_por_ciudad

city  reproducciones
0  Shelbyville           18512
1  Springfield           42741

Comenta tus observaciones aquí:

La cantidad de canciones reproducidas en Springfield es más del doble que la cantidad de canciones escuchadas en Shelbyville.


Ahora agrupemos los datos por día de la semana y encontremos el número de canciones reproducidas el lunes, miércoles y viernes. Utiliza el mismo método que antes, pero ahora necesitamos una agrupación diferente.

In [47]:
# Calcular las canciones reproducidas en cada uno de los tres días
canciones_por_dia=df.groupby(by='day')['track'].count().reset_index(name='reproducciones')
canciones_por_dia

day  reproducciones
0     Friday           21840
1     Monday           21354
2  Wednesday           18059

Comenta tus observaciones aquí:

Lunes y Viernes son los días que mas se reproducen canciones.

Ya sabes cómo contar entradas agrupándolas por ciudad o día. Ahora necesitas escribir una función que pueda contar entradas según ambos criterios simultáneamente.

Crea la función number_tracks() para calcular el número de canciones reproducidas en un determinado día y ciudad. La función debe aceptar dos parámetros:

day: un día de la semana para filtrar. Por ejemplo, 'Monday' (lunes).
city: una ciudad para filtrar. Por ejemplo, 'Springfield'.
Dentro de la función, aplicarás un filtrado consecutivo con indexación lógica.

Primero filtra los datos por día y luego filtra la tabla resultante por ciudad.

Después de filtrar los datos por dos criterios, cuenta el número de valores de la columna 'user_id' en la tabla resultante. Este recuento representa el número de entradas que estás buscando. Guarda el resultado en una nueva variable y devuélvelo desde la función.

In [48]:
# Declara la función number_tracks() con dos parámetros: day= y city=.
def number_tracks(day,city):
    

    # Almacena las filas del DataFrame donde el valor en la columna 'day' es igual al parámetro day=
    df_x=df[df['day']==day]
    # Filtra las filas donde el valor en la columna 'city' es igual al parámetro city=
    df_x=df_x[df_x['city']==city]
    # Extrae la columna 'user_id' de la tabla filtrada y aplica el método count()
    track_list_count=df_x['user_id'].count()
    # Devolve el número de valores de la columna 'user_id'
    return track_list_count

Llama a number_tracks() seis veces, cambiando los valores de los parámetros para que recuperes los datos de ambas ciudades para cada uno de los tres días.

In [49]:
# El número de canciones reproducidas en Springfield el lunes
#track_list_count=df[(df['day']=='Monday')&(df['city']=='Springfield')]['user_id'].count()

track_list_count=number_tracks(day='Monday',city='Springfield')
print(f"El número de canciones reproducidas en Springfield, el lunes, es: {track_list_count}")

El número de canciones reproducidas en Springfield, el lunes, es: 15740


In [50]:
# El número de canciones reproducidas en Shelbyville el lunes
#track_list_count=df[(df['day']=='Monday')&(df['city']=='Shelbyville')]['user_id'].count()
track_list_count=number_tracks(day='Monday',city='Shelbyville')
print(f"El número de canciones reproducidas en Shelbyville, el lunes, es: {track_list_count}")

El número de canciones reproducidas en Shelbyville, el lunes, es: 5614


In [51]:
# El número de canciones reproducidas en Springfield el miércoles
#track_list_count=df[(df['day']=='Wednesday')&(df['city']=='Springfield')]['user_id'].count()
track_list_count=number_tracks(day='Wednesday',city='Springfield')
print(f"El número de canciones reproducidas en Springfield, el miércoles, es: {track_list_count}")

El número de canciones reproducidas en Springfield, el miércoles, es: 11056


In [52]:
# El número de canciones reproducidas en Shelbyville el miércoles
#track_list_count=df[(df['day']=='Wednesday')&(df['city']=='Shelbyville')]['user_id'].count()
track_list_count=number_tracks(day='Wednesday',city='Shelbyville')
print(f"El número de canciones reproducidas en Shelbyville, el miércoles, es: {track_list_count}")

El número de canciones reproducidas en Shelbyville, el miércoles, es: 7003


In [53]:
# El número de canciones reproducidas en Springfield el viernes
#track_list_count=df[(df['day']=='Friday')&(df['city']=='Springfield')]['user_id'].count()
track_list_count=number_tracks(day='Friday',city='Springfield')
print(f"El número de canciones reproducidas en Springfield, el viernes, es: {track_list_count}")

El número de canciones reproducidas en Springfield, el viernes, es: 15945


In [54]:
# El número de canciones reproducidas en Shelbyville el viernes
#track_list_count=df[(df['day']=='Friday')&(df['city']=='Shelbyville')]['user_id'].count()
track_list_count=number_tracks(day='Friday',city='Shelbyville')
print(f"El número de canciones reproducidas en Shelbyville, el viernes, es: {track_list_count}")

El número de canciones reproducidas en Shelbyville, el viernes, es: 5895


### **Conclusiones**

Comenta si la hipótesis es correcta o se debe rechazar. Explica tu razonamiento.

### **Hipótesis: La actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad.**

Con base a la información entregada y procesada, se puede confirmar que la hipótesis es correcta y podemos aceptarla. A continuación presento el resumen del análisis con lo que respaldo mi razonamiento:

El número de canciones reproducidas en Springfield, el lunes, es: 15740

El número de canciones reproducidas en Springfield, el miércoles, es: 11056

El número de canciones reproducidas en Springfield, el viernes, es: 15945

El número de canciones reproducidas en Shelbyville, el lunes, es: 5614

El número de canciones reproducidas en Shelbyville, el miércoles, es: 7003

El número de canciones reproducidas en Shelbyville, el viernes, es: 5895

Habitantes Shelbyville 4,674 (Nota1) Habitantes Springfield: 112,544 (Nota 2)

Notas.

Nota1: Considerando dato del censo correspondiente al 2020 (Wikipedia) y suponiendo que es la ciudad ubicada en el estado de Illinois.

Nota2: (Fuente: Wikipedia)

## **Conclusiones generales**

Resume aquí tus conclusiones sobre la hipótesis.

A continuación, presento mis observaciones sobre el presente análisis de datos y la conclusión sobre la hipótesis:

1. Estamos comparando dos ciudades muy distintas en cuanto a cantidad de población. Springfield supera por 24x a Shelbyville.

2. Mientras que Springfield el miércoles es el día que hay menos reproducciones, en Shelbyville es el día que hay más.

3. El miércoles en Springfield las reproducciones BAJAN UN 43% vs el promedio de los dos días con más reproducciones (lunes y viernes).

4. El miércoles en Shelbyville las reproducciones SUBEN UN 22% vs el promedio de los dos días con más reproducciones (lunes y viernes).

Es con estos puntos presentados, que la información con la que se trabajó, arroja suficiente evidencia para sugerir que la actividad de usuarios y usuarias difiere según el día de la semana y dependiendo la ciudad.